### ISSUE TITLE
File ownership issue on Linux

### ISSUE THREAD

**Issue by vuldin** [redpanda-data-university/rp-102-stream-processing#2]
There is an issue when running through steps in the following link: https://university.redpanda.com/courses/hands-on-redpanda-stream-processing

Files are not readable (and some files are also not readable) by the user starting with the following command at the beginning of the `Enabling Redpanda Data Transforms` section:

```bash
rpk wasm generate mask-card
cd transforms/mask-card
```

This file generates a node project with `src/main.js` and `package.json` having `600` permissions. This is a separate issue in itself, but resolving this wouldn't completely close this ticket.

Since these files need to be readable/writable by both the user and the user with UID `101` (redpanda), there are two solutions:
1. Make the files readable/writable by everyone
2. Add both users to a shared group, change the group owner to that group, and make files readable/writable by owner group

Option 2 isn't a great option though, as the UID `101` user is not actually the `redpanda` user on most systems. On my system for example:

```bash
> id -un 101 
systemd-resolve
```

So the only option I can think of is the following:
```bash
sudo chmod -R 777 {configs,transforms}
```

**Comment by mitch-seymour** [comment]
We had previously added the following disclaimer to the course in order to address a similar issue (https://github.com/redpanda-data-university/rp-102-stream-processing/issues/1):

> **Linux users only**
If you are on a Linux machine, you will need to update the file ownership first:
`sudo chown -R 101:101 {configs,transforms}`

I understand that uid `101` may not work if running outside of the container, but the tutorial deploys this inside the running container where uid `101` corresponds to the `redpanda` user, so it should work (and appeared to resolve the other issue that was opened). But perhaps if people run that before updating the transform code locally, that could present an issue. Is that what you are experiencing? And does running the `chown` as the very last step before `rpk wasm deploy --name mask-card mask-card/dist/main.js` fix the issue in your env?

Anyways, thanks for surfacing this.

**Comment by tombnorwood** [comment]
I've run into the same issue, and it does not appear to me that running `sudo chown -R 101:101 {configs,transforms}`
only after all other steps but before running `rpk wasm deploy --name mask-card mask-card/dist/main.js` fully solves the problem.

When I tried to go this route on my system (Ubuntu 20.04 LTS), when I try to run the container as configured without changing ownership on the `configs/` directory, I get the following error after `running docker compose logs`:  
```
redpanda-1  | Error: error writing to temporary file: open /etc/redpanda/redpanda-20221201133249.yaml: permission denied
```
and the container never runs. If the container isn't running, the call to `rpk wasm generate mask-card` fails.

**Comment by vinay-sg** [comment]
Can confirm that the issue still persists even after setting permissions for configs and transforms. the error message is also similar to the one faced by @tombnorwood. and the container finally exits.
